server/executor/poller_executor.go

```diff
@@ -98,17 +102,46 @@ func (pe DefaultPollerExecutor) ExecuteRequest(request *PollingRequest) (bool, m
 		return false, model.Run{}, err
 	}
 
+	if request.IsFirstRequest() {
+		connectionResult := traceDB.TestConnection(request.ctx)
```

Comment:
Team, a philosophical question here: it makes sense to stop the polling process here if we receive a connection error on this test?